dot/peerset/peerset.go

```diff
@@ -539,15 +574,15 @@ func (ps *PeerSet) removePeer(setID int, peers ...peer.ID) error {
 			// disconnect and forget
 			err := ps.peerState.disconnect(setID, pid)
 			if err != nil {
-				return err
+				return fmt.Errorf("cannot disconnect peer %s at set %d: %w", pid, setID, err)
 			}
 
 			if err = ps.peerState.forgetPeer(setID, pid); err != nil {
-				return err
+				return fmt.Errorf("cannot forget peer %s from set %d: %w", pid, setID, err)
 			}
 		} else if status == notConnectedPeer {
 			if err := ps.peerState.forgetPeer(setID, pid); err != nil {
-				return err
+				return fmt.Errorf("cannot forget peer %s from set %d: %w", pid, setID, err)
```

Comment:
pid and setid error context should be added by the `forgetPeer` method and not here